Resources/js/router.js

```diff
@@ -1,222 +1,241 @@
-goog.provide('fos.Router');
-
-goog.require('goog.structs.Map');
-goog.require('goog.array');
-goog.require('goog.object');
-goog.require('goog.uri.utils');
+'use strict';
 
 /**
- * @constructor
- * @param {fos.Router.Context=} opt_context
- * @param {Object.<string, fos.Router.Route>=} opt_routes
+ * @fileoverview This file defines the Router class.
+ *
+ * You can compile this file by running the following command from the Resources folder:
+ *
+ *    npm run build
  */
-fos.Router = function(opt_context, opt_routes) {
-    this.context_ = opt_context || {base_url: '', prefix: '', host: '', scheme: ''};
-    this.setRoutes(opt_routes || {});
-};
-goog.addSingletonGetter(fos.Router);
 
 /**
- * @typedef {{
- *     tokens: (Array.<Array.<string>>),
- *     defaults: (Object.<string, string>),
- *     requirements: Object,
- *     hosttokens: (Array.<string>)
- * }}
+ * Router singleton.
+ * @type {null|Router}
  */
-fos.Router.Route;
+let instance = null;
 
-/**
- * @typedef {{
- *     base_url: (string)
- * }}
- */
-fos.Router.Context;
+class Router {
 
-/**
- * @param {Object.<string, fos.Router.Route>} routes
- */
-fos.Router.prototype.setRoutes = function(routes) {
-    this.routes_ = new goog.structs.Map(routes);
-};
+    /**
+     * @constructor
+     * @param {Router.Context=} context
+     * @param {Object.<string, Router.Route>=} routes
+     */
+    constructor(context, routes) {
+        instance = this;
```

Comment:
this is wrong. The old code was still allowing to create multiple instance of the router, when using the constructor rather than using the singleton getter.. The router itself is not a singleton